dataflow/src/org/checkerframework/dataflow/analysis/FlowExpressions.java

```diff
@@ -478,7 +480,9 @@ public boolean equals(Object obj) {
                 return false;
             }
             LocalVariable other = (LocalVariable) obj;
-            return other.element.equals(element);
+            return other.element.name.toString().equals(element.name.toString()) &&
+                   other.element.type.toString().equals(element.type.toString()) &&
+                   other.element.owner.toString().equals(element.owner.toString());
```

Comment:
I couldn't find any code in javac that compares two Symbols.. strange, but maybe I've missed something obvious.
The only thing I found is com.sun.tools.javac.jvm.Pool.Variable.equals(Object) which only compares the name and owner, ignoring the type.